tests/quote_props_typescript/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -0,0 +1,94 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`types.ts 1`] = `
+====================================options=====================================
+parsers: ["typescript"]
+printWidth: 80
+quoteProps: "as-needed"
+                                                                                | printWidth
+=====================================input======================================
+type T = {
+  0: string;
+  5: number;
+}
+
+type U = {
+  0: string;
+  "5": number;
+}
+
+=====================================output=====================================
+type T = {
+  0: string;
+  5: number;
+};
+
+type U = {
+  0: string;
+  "5": number;
+};
+
+================================================================================
+`;
+
+exports[`types.ts 2`] = `
+====================================options=====================================
+parsers: ["typescript"]
+printWidth: 80
+quoteProps: "preserve"
+                                                                                | printWidth
+=====================================input======================================
+type T = {
+  0: string;
+  5: number;
+}
+
+type U = {
+  0: string;
+  "5": number;
+}
+
+=====================================output=====================================
+type T = {
+  0: string;
+  5: number;
+};
+
+type U = {
+  0: string;
+  "5": number;
+};
+
+================================================================================
+`;
+
+exports[`types.ts 3`] = `
+====================================options=====================================
+parsers: ["typescript"]
+printWidth: 80
+quoteProps: "consistent"
+                                                                                | printWidth
+=====================================input======================================
+type T = {
+  0: string;
+  5: number;
+}
+
+type U = {
+  0: string;
+  "5": number;
+}
+
+=====================================output=====================================
+type T = {
+  0: string;
+  5: number;
+};
+
+type U = {
+  0: string;
```

Comment:
Shouldn't this be quoted, since we're using `consistent`? Or do these have different semantics in typescript?

Reply:
Looks like `0` and `"0"` are interchangeable here: https://typescript-play.js.org/#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXygGdCQYMAVATwAcQAecgPgAoAPALnnIEpOA3HFmAAoYRhoIAYjhzwAvPADe8AAycARBhCEM6+AF9hoSLARg8O+B3jScwoiTJVadTdt0tWAbRUBdbvbEpBQSrlo66p5e6irq-qLitPAAQrDySvAx6hrhugZG4NBw8OaolpScqTCBjiEubhEslD7xDsHO9A0ezM1Z8UA

Reply:
https://github.com/prettier/prettier/pull/5934#discussion_r261809360

I'm not sure why this line exists. Will do some digging.

Reply:
Found it - #4516 - Seems to be the correct behaviour.

Reply:
Ah, okay, cool. Thanks!